src/transformers/modeling_utils.py

```diff
@@ -1605,11 +1620,14 @@ def _autoset_attn_implementation(
                     if not isinstance(requested_attn_implementation, dict)
                     else requested_attn_implementation.get(key, None)
                 )
+                if curr_attn_implementation is None:
+                    config._attn_implementation_autoset = True
```

Comment:
The actual change (2): if the user DOESN'T set attn_implementation in a sub-model, then it was automatically inferred

Reply:
`py.test tests/models/ -k test_attn_implementation_composite_models` and `py.test tests/models/ -k test_sdpa_can_dispatch_composite_models` all green ✅